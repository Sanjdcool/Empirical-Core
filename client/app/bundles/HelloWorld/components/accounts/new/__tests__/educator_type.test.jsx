import React from 'react';
import { shallow } from 'enzyme';

import EducatorType from '../educator_type';

import UsK12View from '../../school/us_k12_view'
import NotUsK12View from '../../school/not_us_k12_view'

describe('EducatorType component', () => {

  it('should render if stage is 1', () => {
      const wrapper = shallow(
        <EducatorType />
      );
      expect(wrapper).toMatchSnapshot();
  });

  it('should render UsK12View if stage state is 2', () => {
      const wrapper = shallow(
        <EducatorType />
      );
      wrapper.setState({ stage: 2 });
      expect(wrapper.find(UsK12View)).toHaveLength(1);
  });

  it('should render NotUsK12View if stage state is 3', () => {
      const wrapper = shallow(
        <EducatorType />
      );
      wrapper.setState({ stage: 3 });
      expect(wrapper.find(NotUsK12View)).toHaveLength(1);
  });

});
