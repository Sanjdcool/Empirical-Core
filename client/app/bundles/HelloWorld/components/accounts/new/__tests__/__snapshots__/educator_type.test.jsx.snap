// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EducatorType component should render if stage is 1 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="educator-type"
>
    <h3>
        Are you a faculty member at a U.S K-12 School?*
    </h3>
    <div
        className="option-wrapper"
    >
        <button
            className="button-green"
            onClick={[Function]}
        >
            Yes
        </button>
        <button
            className="button-green"
            onClick={[Function]}
        >
            No
        </button>
    </div>
    <div>
        *K-12 is a term for school grades prior to college.
        <br />
        These grades span from kindergarten through the 12th grade.
    </div>
</div>,
  "nodes": Array [
    <div
      className="educator-type"
>
      <h3>
            Are you a faculty member at a U.S K-12 School?*
      </h3>
      <div
            className="option-wrapper"
      >
            <button
                  className="button-green"
                  onClick={[Function]}
            >
                  Yes
            </button>
            <button
                  className="button-green"
                  onClick={[Function]}
            >
                  No
            </button>
      </div>
      <div>
            *K-12 is a term for school grades prior to college.
            <br />
            These grades span from kindergarten through the 12th grade.
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <educator_type />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Constructor {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "finish": [Function],
        "goToStage": [Function],
        "props": Object {},
        "refs": Object {},
        "selectSchool": [Function],
        "state": Object {
          "stage": 1,
        },
        "stateSpecificComponents": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="educator-type"
>
          <h3>
                    Are you a faculty member at a U.S K-12 School?*
          </h3>
          <div
                    className="option-wrapper"
          >
                    <button
                              className="button-green"
                              onClick={[Function]}
                    >
                              Yes
                    </button>
                    <button
                              className="button-green"
                              onClick={[Function]}
                    >
                              No
                    </button>
          </div>
          <div>
                    *K-12 is a term for school grades prior to college.
                    <br />
                    These grades span from kindergarten through the 12th grade.
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="educator-type"
>
          <h3>
                    Are you a faculty member at a U.S K-12 School?*
          </h3>
          <div
                    className="option-wrapper"
          >
                    <button
                              className="button-green"
                              onClick={[Function]}
                    >
                              Yes
                    </button>
                    <button
                              className="button-green"
                              onClick={[Function]}
                    >
                              No
                    </button>
          </div>
          <div>
                    *K-12 is a term for school grades prior to college.
                    <br />
                    These grades span from kindergarten through the 12th grade.
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <educator_type />,
}
`;
